File: tests/simple_class/PasswordValidator.swift
class PasswordValidator {
    private let minLength: Int
    private let maxLength: Int
    private let requireUppercase: Bool
    private let requireLowercase: Bool
    private let requireDigit: Bool
    private let requireSpecialChar: Bool
    private let forbiddenChars: Set<Character>

    private init(builder: Builder) {
        self.minLength = builder.minLength
        self.maxLength = builder.maxLength
        self.requireUppercase = builder.requireUppercase
        self.requireLowercase = builder.requireLowercase
        self.requireDigit = builder.requireDigit
        self.requireSpecialChar = builder.requireSpecialChar
        self.forbiddenChars = builder.forbiddenChars
    }

    enum ValidationResult {
        case valid
        case invalid([String])
    }

    func validate(_ password: String) -> ValidationResult {
        var errors = [String]()

        if password.count < minLength { errors.append("Password is too short") }
        if password.count > maxLength { errors.append("Password is too long") }
        if requireUppercase && !password.contains(where: { $0.isUppercase }) { errors.append("Missing uppercase letter") }
        if requireLowercase && !password.contains(where: { $0.isLowercase }) { errors.append("Missing lowercase letter") }
        if requireDigit && !password.contains(where: { $0.isNumber }) { errors.append("Missing digit") }
        if requireSpecialChar && !password.contains(where: { !$0.isLetter && !$0.isNumber }) { errors.append("Missing special character") }
        if password.contains(where: { forbiddenChars.contains($0) }) { errors.append("Contains forbidden character") }

        return errors.isEmpty ? .valid : .invalid(errors)
    }

    class Builder {
        var minLength: Int = 8
        var maxLength: Int = 64
        var requireUppercase: Bool = false
        var requireLowercase: Bool = false
        var requireDigit: Bool = false
        var requireSpecialChar: Bool = false
        var forbiddenChars: Set<Character> = []

        func setMinLength(_ length: Int) -> Builder {
            self.minLength = length
            return self
        }

        func setMaxLength(_ length: Int) -> Builder {
            self.maxLength = length
            return self
        }

        func setRequireUppercase(_ require: Bool) -> Builder {
            self.requireUppercase = require
            return self
        }

        func setRequireLowercase(_ require: Bool) -> Builder {
            self.requireLowercase = require
            return self
        }

        func setRequireDigit(_ require: Bool) -> Builder {
            self.requireDigit = require
            return self
        }

        func setRequireSpecialChar(_ require: Bool) -> Builder {
            self.requireSpecialChar = require
            return self
        }

        func setForbiddenChars(_ chars: Set<Character>) -> Builder {
            self.forbiddenChars = chars
            return self
        }

        func build() throws -> PasswordValidator {
            guard minLength > 0 else {
                throw ValidationError.invalidMinLength
            }
            guard maxLength >= minLength else {
                throw ValidationError.invalidMaxLength
            }
            return PasswordValidator(builder: self)
        }

        enum ValidationError: Error {
            case invalidMinLength
            case invalidMaxLength
        }
    }

    static func build(_ configuration: (Builder) -> Void) throws -> PasswordValidator {
        let builder = Builder()
        configuration(builder)
        return try builder.build()
    }
}

File: tests/simple_class/password_validator.dart
import 'dart:core';

enum ValidationResult {
    valid,
    invalid,
}

class PasswordValidator {
  final int minLength;
  final int maxLength;
  final bool requireUppercase;
  final bool requireLowercase;
  final bool requireDigit;
  final bool requireSpecialChar;
  final Set<String> forbiddenChars;

  PasswordValidator._({
    required this.minLength,
    required this.maxLength,
    required this.requireUppercase,
    required this.requireLowercase,
    required this.requireDigit,
    required this.requireSpecialChar,
    required this.forbiddenChars,
  });

  ValidationResult validate(String password) {
    List<String> errors = [];

    if (password.length < minLength) errors.add("Password is too short");
    if (password.length > maxLength) errors.add("Password is too long");
    if (requireUppercase && !password.contains(RegExp(r'[A-Z]'))) errors.add("Missing uppercase letter");
    if (requireLowercase && !password.contains(RegExp(r'[a-z]'))) errors.add("Missing lowercase letter");
    if (requireDigit && !password.contains(RegExp(r'\d'))) errors.add("Missing digit");
    if (requireSpecialChar && !password.contains(RegExp(r'[!@#$%^&*(),.?":{}|<>]'))) errors.add("Missing special character");
    if (password.split('').any((char) => forbiddenChars.contains(char))) errors.add("Contains forbidden character");

    return errors.isEmpty ? ValidationResult.valid : ValidationResult.invalid;
  }

  static Builder builder() => Builder();

  static Future<PasswordValidator> build(Function(Builder) configuration) async {
    final builder = Builder();
    configuration(builder);
    return builder.build();
  }
}

class Builder {
  int minLength = 8;
  int maxLength = 64;
  bool requireUppercase = false;
  bool requireLowercase = false;
  bool requireDigit = false;
  bool requireSpecialChar = false;
  Set<String> forbiddenChars = {};

  Builder setMinLength(int length) {
    minLength = length;
    return this;
  }

  Builder setMaxLength(int length) {
    maxLength = length;
    return this;
  }

  Builder setRequireUppercase(bool require) {
    requireUppercase = require;
    return this;
  }

  Builder setRequireLowercase(bool require) {
    requireLowercase = require;
    return this;
  }

  Builder setRequireDigit(bool require) {
    requireDigit = require;
    return this;
  }

  Builder setRequireSpecialChar(bool require) {
    requireSpecialChar = require;
    return this;
  }

  Builder setForbiddenChars(Set<String> chars) {
    forbiddenChars = chars;
    return this;
  }

  Future<PasswordValidator> build() async {
    if (minLength <= 0) {
      throw ValidationError('Invalid minimum length');
    }
    if (maxLength < minLength) {
      throw ValidationError('Invalid maximum length');
    }
    return PasswordValidator._(
      minLength: minLength,
      maxLength: maxLength,
      requireUppercase: requireUppercase,
      requireLowercase: requireLowercase,
      requireDigit: requireDigit,
      requireSpecialChar: requireSpecialChar,
      forbiddenChars: forbiddenChars,
    );
  }
}

class ValidationError implements Exception {
  final String message;
  ValidationError(this.message);

  @override
  String toString() => 'ValidationError: $message';
}